horust/src/horust/formats/service.rs

```diff
@@ -624,6 +631,74 @@ impl From<TerminationSignal> for Signal {
     }
 }
 
+#[derive(Serialize, Clone, Deserialize, Debug, PartialEq)]
+#[serde(rename_all = "kebab-case", deny_unknown_fields)]
+pub struct ResourceLimit {
+    #[serde(default)]
+    /// The CPU time that the process can use, non-positive value means no limit
+    pub(crate) cpu: f64,
+    #[serde(default, skip_serializing, deserialize_with = "str_to_optional_bytes")]
+    /// The maximum amount of memory that the process can use
+    pub(crate) memory: Option<u64>,
+    #[serde(default)]
+    /// The maximum number of processes/threads that the process can create
+    pub(crate) pids_max: Option<u64>,
+}
+
+impl Default for ResourceLimit {
+    fn default() -> Self {
+        ResourceLimit {
+            cpu: -1.0,
+            memory: None,
+            pids_max: None,
+        }
+    }
+}
+
+impl Eq for ResourceLimit {}
```

Comment:
is there a reason why we cannot derive Eq?

Reply:
`f64` doesn't support auto derive Eq.